src/util/providers/AuthProvider.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { createContext, useState, useContext, useEffect } from 'react';
+import ProtonService from '../services/proton.service';
+import firebaseService from '../services/firebase.service';
+
+interface User {
+  actor: string;
+  permission: string;
+  avatar: string;
+  createdAt: Date;
+  name: string;
+  isMember: boolean;
+  memberLevel: string;
+}
+
+interface AuthResponse {
+  success: boolean;
+  user?: any;
+  error?: any;
+}
+
+interface AuthContext {
+  currentUser: User | null;
+  authenticate?: () => Promise<AuthResponse>;
+  signout?: () => void;
+  updateMember?;
+}
+
+const authContext = createContext<AuthContext>({
+  currentUser: null,
+});
+
+export const useAuthContext = () => {
+  const {
+    currentUser,
+    authenticate,
+    signout,
+    updateMember,
+  } = useContext(authContext);
+
+  return {
+    currentUser,
+    authenticate,
+    signout,
+    updateMember,
+  };
+};
+
+const AuthProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const initialUser = JSON.parse(localStorage.getItem('AUTH_USER'));
+    if (initialUser) {
+      ProtonService.restoreSession();
+      setCurrentUser(initialUser);
+    } else {
+      setCurrentUser(null);
+    }
+  }, []);
+
+  const authenticate = async (): Promise<AuthResponse> => {
+    try {
+      let user = await ProtonService.login();
+
+      if (!user) {
+        throw new Error();
+      }
+
+      const query = await firebaseService
+        .collection('members')
+        .where("user", "==", user.actor)
+        .get();
+
+      if (!query.empty) {
+        let member;
+        query.forEach((doc) => {
+          member = doc.data();
+        });
+        user.isMember = true;
+        user.memberLevel = member.level;
+      }
+
+      localStorage.setItem('AUTH_USER', JSON.stringify(user));
```

Comment:
Can you change the key that this is being saved under to AUTH_USER_PROTON_VIP?

Reply:
Oops, sorry I forgot this update! Just made the changes.